uniswap-v2-core/contracts/UniswapV2Pair.sol

```diff
@@ -85,20 +118,31 @@ contract UniswapV2Pair is IUniswapV2Pair, UniswapV2ERC20 {
         emit Sync(reserve0, reserve1);
     }
 
-    // if fee is on, mint liquidity equivalent to 1/6th of the growth in sqrt(k)
+    function _calcFee(uint _sqrtNewK, uint _sqrtOldK, uint _platformFee, uint _circulatingShares) internal pure returns (uint _sharesToIssue) {
+        // Assert newK & oldK        < uint112
+        // Assert _platformFee       < FEE_ACCURACY
+        // Assert _circulatingShares < uint112
+
+        uint256 _scaledGrowth = (_sqrtNewK * ACCURACY) / _sqrtOldK;                      // ASSERT: < UINT256
+        uint256 _scaledMultiplier = ACCURACY - (SQUARED_ACCURACY / _scaledGrowth);          // ASSERT: < UINT128
+        uint256 _scaledTargetOwnership = _scaledMultiplier * _platformFee / FEE_ACCURACY; // ASSERT: < UINT144 during maths, ends < UINT128
+
+        _sharesToIssue = _scaledTargetOwnership.mul(_circulatingShares) / (ACCURACY.sub(_scaledTargetOwnership)); // ASSER: _scaledTargetOwnership < ACCURACY
+    }
+
     function _mintFee(uint112 _reserve0, uint112 _reserve1) private returns (bool feeOn) {
-        address feeTo = IUniswapV2Factory(factory).feeTo();
-        feeOn = feeTo != address(0);
-        uint _kLast = kLast; // gas savings
+        feeOn = platformFee > 0;
+        uint _kLast = kLast;
+
         if (feeOn) {
-            if (_kLast != 0) {
-                uint rootK = Math.sqrt(uint(_reserve0).mul(_reserve1));
-                uint rootKLast = Math.sqrt(_kLast);
-                if (rootK > rootKLast) {
-                    uint numerator = totalSupply.mul(rootK.sub(rootKLast));
-                    uint denominator = rootK.mul(5).add(rootKLast);
-                    uint liquidity = numerator / denominator;
-                    if (liquidity > 0) _mint(feeTo, liquidity);
+            address feeTo = IUniswapV2Factory(factory).feeTo();
```

Comment:
Decide if we leave as `feeTo` because that's what it was, or move to `_feeTo` because its a local variable. Unsure of what Uniswap's convention is and whether they themselves are following it

Reply:
Unfortunately - just from reading our initial check-in (I didn't check original source) - if they are consistent, I have a feeling that uniswap might only use _ when there is an internal member var of the same name... either that or they are just inconsistent.

Reply:
We should discuss - despite their std, I think it would be preferable to prefix all args & local vars...

Reply:
Yep it does seem like `_` is more of a hack to let them redeclare variables locally. So either because they want a copy of the old value or they want to avoid loading the old value from storage too many times.